velodyne_pointcloud/src/lib/rawdata.cc

```diff
@@ -340,7 +341,22 @@ namespace velodyne_rawdata
       // Calculate difference between current and next block's azimuth angle.
       azimuth = (float)(raw->blocks[block].rotation);
       if (block < (BLOCKS_PER_PACKET-1)){
-        azimuth_diff = (float)((36000 + raw->blocks[block+1].rotation - raw->blocks[block].rotation)%36000);
+	raw_azimuth_diff = raw->blocks[block+1].rotation - raw->blocks[block].rotation;
+        azimuth_diff = (float)((36000 + raw_azimuth_diff)%36000);
+	// some packets contain an angle overflow where azimuth_diff < 0 
+	if(raw_azimuth_diff < 0)//raw->blocks[block+1].rotation - raw->blocks[block].rotation < 0)
```

Comment:
What's the purpose of this part? I don't think there will be an angle overflow here.

@sts-thm @JWhitleyAStuff

Reply:
This pull request fixed the issue linked above. Please refer to https://github.com/ros-drivers/velodyne/issues/174#issue-334135152. If you do still have any questions feel free to ask away.

Reply:
Thanks for reply. I understand the change of driver.cc, which you mentioned in the issue. But I couldn’t understand the change of rawdata.cc. It will be nice if you can explain a little, thanks.

Reply:
The packages of the velodyne in certain modes specify the azimuth only for every second group of points (16 or 32, depending on the velodyne). The azimuth update for the group of points in between is calculated as half of the difference between the previous group's azimuth and the next group's azimuth. However if the azimuth 0° is inside your package the difference will be the reflex angle (> 180°) and not its complement, which we would need.

Reply:
In the case azimuth 0° is inside the packet, let's say the previous group's azimuth is 359°, the next group's azimuth is 1°.

`azimuth_diff = (float)((36000 + raw->blocks[block+1].rotation - raw->blocks[block].rotation)%36000);` is 2°, which I think is what we need?